client/api.go

```diff
@@ -169,7 +185,7 @@ func (c *Client) DeleteMessages(messages *Messages, seek *int) error {
 				// An example of an action is a call request.
 				if msg.Type == UserMessage {
 					log.Infof("Deleting message %v from channel %v", msg.ID, msg.ChannelID)
-					if c.dryRun {
+					if c.dryRun || c.SkipChannel(msg.ChannelID) {
```

Comment:
This will technically work but it will cause the entire list of messages to be retrieved for a channel unnecessarily. You could think about adding a condition somewhere in the `PartialDelete()` function to skip a channel.

Reply:
In the `PartialDelete()` function, only direct message channels IDs are fetched, so can't skip guild channels via ID and when fetching messages from a guild, only the guild ID is used. Any other ideas?

Reply:
Changed up stuff a little, now you can pass guild IDs and direct message channel IDs so guilds and DMs can be skipped. But not individual channels in guilds, i guess we could iterate over the `results` in `func (c *Client) GuildMessages()` and remove results that match the skipped channel IDs, but even then we still have to fetch the whole history

Reply:
So now guilds/DMs/individual channels can all be skipped with the `--skip` flag, but it seems a little hacky since individual message's channel IDs are still checked. Fetching of message history is skipped only for full guilds/DMs.